docs/vsphere-golden-image.md

```diff
@@ -0,0 +1,191 @@
+describe by the [Microsoft documentation.](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_keymanagement#administrative-user) 
+
+Alternatively, a PowerShell script [install-openssh.ps1](vsphere_ci/scripts/install-openssh.ps1) is provided to 
+automate the process of installing and configuring the OpenSSH Server with the public key file. For example, you can 
+the following command:
+
+```
+    ./install-openssh.ps1 <path/to/public_key_file>
+```
+
+where, `<path/to/public_key_file>` is the path to the public key file corresponding to the [private key.](../README.md#create-a-private-key-secret)
+
+## 4. Set up the container runtime
+
+Install the `docker` container runtime following the [Microsoft's documentation](https://docs.microsoft.com/en-us/virtualization/windowscontainers/quick-start/set-up-environment?tabs=Windows-Server).
+
+Alternatively, you can run the provided PowerShell script [install-docker.ps1](vsphere_ci/scripts/install-docker.ps1) 
+to programmatically install `docker` in the Windows VM.
+
+## 5. Set up incoming connection for container logs
+
+Create a new firewall rule in the Windows VM to allow incoming connections for container logs, usually 
+on TCP port `10250` by running the following PowerShell command:
+
+```powershell
+    New-NetFirewallRule -DisplayName "ContainerLogsPort" -LocalPort 10250 -Enabled True -Direction Inbound -Protocol TCP -Action Allow -EdgeTraversalPolicy Allow
+```
+
+## 6. Generalize the virtual machine installation
+
+To deploy the Windows VM as a reusable image, you have to first generalize the VM removing computer-specific information 
+such as installed drivers. Running the `sysprep` command with a *unattend* answer file generalizes the image and 
+makes it ready for future deployments, maintaining all the changes needed for the WMCO installation. 
+
+You can find an example [unattend.xml](unattend.xml) file in this repository. It is just an example and, it **should 
+not** be used directly. You must customize the example to fit your needs.
+
+To execute the `sysprep` command use:
+
+```cmd
+    C:\Windows\System32\Sysprep\sysprep.exe /generalize /oobe /shutdown /unattend:<path/to/unattend.xml>
+```
+
+where `<path/to/unattend.xml>` is the path to the customized answer file.
+
+Note: There is [a limit](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/sysprep--generalize--a-windows-installation#limits-on-how-many-times-you-can-run-sysprep)
+on how many times you can run the `sysprep` command.
+
+## 7. Set up the virtual machine golden image
+
+Once the `sysprep` command completes the Windows virtual machine will power off. 
+
+You **must not** use such Windows virtual machine anymore. It is recommended to clone the newly created virtual machine 
+before executing the [Step #6](vsphere-golden-image.md#6-generalize-the-virtual-machine-installation) 
+and run the `sysprep` command in the clone. 
```

Comment:
I suggest you remove `It is recommended...` sentence. It actually makes me think I need to clone the VM again before I use in the `MachineSet`